core/Updater/Migration.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+namespace Piwik\Updater;
+
+use Piwik\Db;
+use Exception;
+
+/**
+ * Base class for migrations. Any migration must extend this class.
+ *
+ * @api
+ */
+abstract class Migration
```

Comment:
I first used an `interface` but this would mean we cannot add any new features as every new addition of a new method would break API. By using `abstract` class we can add at least new method like `checkPrecondition` or `isMajorUpdate`.